### DISCUSSION TITLE
How to create a VertexOnlyMesh for interpolating to a array ?

### DISCUSSION THREAD

**Discussion by ho11laqe** [firedrakeproject/firedrake#2714] [Firedrake support] [answered]
Hello,

I want to get an array-shaped interpolation of a Function, but I struggle to create a VertexOnlyMesh. My approach is:
```
square_mesh = UnitSquareMesh(12,10)
coordinates = [[x,y] for x in np.arange(bottom,top,1000) for y in np.arange(left, right, 1000)]
VOM = firedrake.VertexOnlyMesh(square_mesh, vertexcoords=coordinates)
elevation_array= icepack.interpolate(elevation_function, VOM)
```
but the resulting VOM.coordinates.dat.data has the shape (0,2).

Thank you for your help!

**Comment by ho11laqe** [comment, marked as answer]
I used the firedrake FunctionPlotter to get Coordinates and Values from the mesh.
```    
from firedrake import  FunctionPlotter
function_plotter = FunctionPlotter(mesh, 10)
triangulation = function_plotter.triangulation
values = function_plotter(elevation_function)
```

**Comment by ReubenHill** [comment]
Sorry for the delay in getting back to you! 

> VOM.coordinates.dat.data has the shape (0,2).

This suggests that all the points you specified when you did 
```python
coordinates = [[x,y] for x in np.arange(bottom,top,1000) for y in np.arange(left, right, 1000)]
```
were outside the `UnitSquareMesh(12,10)` which is a square mesh which has border vertices `(0, 0)`, `(0, 1)`, `(1, 0)` and `(1, 1)`.

If you want to be warned that that you are losing points on `VertexOnlyMesh` construction write
```
firedrake.VertexOnlyMesh(square_mesh, vertexcoords=coordinates, missing_points_behaviour="warn")
```
or for error write
```
firedrake.VertexOnlyMesh(square_mesh, vertexcoords=coordinates, missing_points_behaviour="error")
```
